syntax = "proto3";

package greet;

// The greeting service definition.
service Greeter
{
    // Sends a greeting.
    rpc SayHello (HelloRequest) returns (HelloReply);

    // Sends a task request.
    rpc RequestTask (TaskRequest) returns (TaskResponse);
}

// The request message containing the user's name.
message HelloRequest
{
    string name = 1;
}

// The response message containing the greeting.
message HelloReply
{
    string message = 1;
}

message TaskRequest
{
    string client_id = 1;
}

message TaskResponse
{
    string message = 1;
}